### PR TITLE
add some sanity checks for cursor positions

### PR DESCRIPTION
**Pull request by steveoc64**
With the entry widgets - there are some issues there with managing the cursor positions after manually updating the text contents.

You can end up with the cursor position pointing to a non-existent location in the new string.

This patch sets the cursor position to be the end of the string after doing a SetText() if the current position is > length of the new string.

### CODE REVIEW

**Comment by andydotxyz** [comment]
This pull request is failing as entry.go appears to be in the root of the project instead of the widget package... Not sure why.
Do you want to try and fix this or should I download and apply the changes myself?

File: widget/entry.go [hunk 1]
```diff
@@ -126,6 +130,13 @@ func (e *Entry) cursorTextPos() int {
 	}
 	pos += e.CursorColumn
 
+	// Some sanity checks here
+	if pos > len(e.Text) {
```
Comment (andydotxyz) [inline widget/entry.go]: If we are off the end should we not set to the end rather than the beginning?

File: driver/efl/loop.go [hunk 2]
```diff
@@ -79,6 +80,7 @@ func runEFL() {
 		case data := <-renderQueue:
 			data.c.dirty[data.co] = true
 		case <-tick.C:
+			//onMain = C.eina_main_loop_is()
```
Comment (andydotxyz) [inline driver/efl/loop.go]: I think this can go...

File: driver/efl/window.go [hunk 3]
```diff
@@ -18,13 +18,16 @@ package efl
 // void force_render();
 import "C"
 
-import "log"
-import "os"
-import "runtime"
-import "strconv"
-import "unsafe"
+import (
```
Comment (steveoc64) [inline driver/efl/window.go]: go 1.11.2 auto-formatting

**Pull request merged**